android/src/main/java/com/swmansion/rnscreens/Screen.kt

```diff
@@ -191,7 +195,20 @@ class Screen(
             val width = r - l
             val height = b - t
 
-            dispatchShadowStateUpdate(width, height, t)
+            if (!insetsApplied && headerConfig?.isHeaderHidden == false && headerConfig?.isHeaderTranslucent == false) {
+                val topLevelDecorView =
+                    requireNotNull(
+                        reactContext.currentActivity?.window?.decorView,
+                    ) { "[RNScreens] DecorView is required for applying inset correction, but was null." }
+
+                val topInset = DecorViewInsetsUtils.getDecorViewTopInset(topLevelDecorView)
+                val correctedHeight = height - topInset
+                val correctedOffsetY = t + topInset
+
+                dispatchShadowStateUpdate(width, correctedHeight, correctedOffsetY)
```

Comment:
This code assumes the top insets won't be consumed along the way right? 

Do we still experience problems in case of `Tabs > Stack > Stack`? I know this is exotic scenario, but just so that we are aware.

Reply:
Okay, the content is not jumping, the inset is not consumed and the header is applying extra padding - this shouldn't be needed for the nested stack to offset the header. Need to test some more scenarios, because when only the nested header is shown, then we need to apply the update for the nested stack. Investigating the solution how to apply the correction to the top-level header only ..

https://github.com/user-attachments/assets/0590ae95-c2e5-4471-92d9-55cb223651eb

Reply:
https://github.com/software-mansion/react-native-screens/pull/3442/commits/cf775238d2d44375560366c8bf8cb7196a8f870a

Reply:
as discussed internally, this shouldn't be part of this PR, I'm reverting it and creating myself a ticket for handling double insets